code/datums/gamemode/factions/bloodcult/bloodcult_spells.dm

```diff
@@ -239,3 +235,62 @@
 		H.visible_message("<span class='warning'>\The [user] squeezes the blood in their hand, and it takes the shape of a dagger!</span>",
 			"<span class='warning'>You squeeze the blood in your hand, and it takes the shape of a dagger.</span>")
 		playsound(H, 'sound/weapons/bloodyslice.ogg', 30, 0,-2)
+
+//SPELL IV
+/spell/cult/arcane_dimension
+	name = "Arcane Dimension (empty)"
+	desc = "Cast while holding an Arcane Tome to discretly store it through the veil."
+	hud_state = "cult_pocket_empty"
+
+	invocation_type = SpI_NONE
+	charge_type = Sp_RECHARGE
+	charge_max = 0
+	range = 0
+	spell_flags = null
+	insufficient_holder_msg = ""
+	still_recharging_msg = ""
+
+	cast_delay = 0
+
+	var/obj/item/weapon/tome/stored_tome = null
+
+/spell/cult/arcane_dimension/choose_targets(var/mob/user = usr)
+	return list(user)
+
+/spell/cult/arcane_dimension/cast(var/list/targets, var/mob/living/carbon/user)
+	..()
+	if (stored_tome)
+		stored_tome.forceMove(get_turf(user))
+		if (user.get_inactive_hand() && user.get_active_hand())//full hands
+			to_chat(user,"<span class='warning'>Your hands being full, your [stored_tome] had nowhere to fall but on the ground.</span>")
+		else
+			to_chat(user,"<span class='notice'>You hold your hand palm up, and your [stored_tome] drops in it from thin air.</span>")
+			user.put_in_hands(stored_tome)
+		stored_tome = null
+		name = "Arcane Dimension (empty)"
+		connected_button.name = name
+		desc = "Cast while holding an Arcane Tome to discretly store it through the veil."
+		hud_state = "cult_pocket_empty"
+		connected_button.overlays.len = 0
+		connected_button.MouseExited()
+		return
+
+	var/obj/item/weapon/tome/held_tome = user.get_active_hand()
```

Comment:
Again, var/obj/item/weapon/tome/held_tome = locate() in user.held_items, then remove the if(!held_tome)